tests/test_metric_output.py

```diff
@@ -0,0 +1,39 @@
+import requests
+import yaml
+
+
+class TestMetricOutput:
+    def test_vcsa_data_comparison(self, setup_vcsa_url):
+        response = requests.get(setup_vcsa_url)
+        metrics = self.process_response_data(response)
+
+        with open('tests/vcsa_output_comparison.yaml', 'r') as fp:
+            comparison_data = yaml.safe_load(fp)
+
+        for key in comparison_data.keys():
+            difference = set(comparison_data[key]).difference(metrics)
+            assert len(difference) == 0
+
+    def test_vcenter_backend_down(self, request, setup_vcsa_url, setup_vcenter):
```

Comment:
This test is not testing how the exporter is interacting against a down backend. setup_vcenter would need another host here and then would need to do the query, wouldn't it?

Reply:
The connection is done with the vcenter name, which gets assigned to the connection target attribute within the constructor of the Connection class. So it does sent a request against a non existing host. The VCSA exporter then returns default metrics created by Prometheus 
Could you elaborate a little bit further?

Reply:
Alright, so it is supposed to just check for the empty prometheus server.
Interesting test case then would be to basically run
* `test_vcsa_data_comparsion`
* set the backend down 
* check ouput
* set backend up
* check output

Reply:
I presume with the point `test_vcsa_data_comparsion` you mean the executed test before test_vcenter_backend_down?
If yes, I would implement it as you said, set backend down, check the output, set the backend up, check the output.

Reply:
I implemented another test for that, called 'test_vcenter_backend_up'. This test checks if expected metrics are returning. The difference to 'test_prometheus_server_reply' is, that it checks if the returning data is correct.

Reply:
That's a valid test case. Can we also add it in one test as described above? 
This then would be more like a real world scenario like backend is up, then it goes down, then it goes up again. Idea is to see, if the code can survive.

Reply:
Refactored those testcase into 'test_backend_recovery_from_down'